### ISSUE TITLE
JPADatabase with GeneratedValue does not account for dialect

### ISSUE THREAD

**Issue by Fathardie** [jOOQ/jOOQ#6332]
Hi!

We have started to experiment with jOOQ and see if it can take care of all our simple ORM use-cases. We have, however, run into a bit of a snag. We are using a PostgreSQL database where the schema completely depends on the JPA annotated entities. So we are using the jOOQ JPADatabase code generator to generate the schema for us. We are using `@GeneratedValue(strategy = GenerationType.IDENTITY)` for the `id` field in a `@MappedSuperclass`. This results in the generation of a `id` column in the table. Due to the `@GeneratedValue`, there is also a default value included:

```
....defaultValue(org.jooq.impl.DSL.field("(NEXT VALUE FOR PUBLIC.SYSTEM_SEQUENCE_20F10531_7EEF_4550_A013_1C8EDE599D2D)", org.jooq.impl.SQLDataType.BIGINT)
```
While this seems to be correct syntax for TransactSQL, it is not for PostgreSQL:
```
https://docs.microsoft.com/en-us/sql/t-sql/functions/next-value-for-transact-sql
https://www.postgresql.org/docs/9.6/static/functions-sequence.html
```
Not withstanding that this SEQUENCE table does not even exist. This is of course completely dependent on how Hibernate implements these generated values but currently it breaks our pipeline.

To us this looks like a bug but as we are not completely familiar with SQL and jOOQ yet, it might also be wrong usage. Any help is appreciated!

**Comment by lukaseder** [comment]
Thank you very much for reporting. I believe this is a known issue but I cannot seem to find the issue on GitHub.

The reason for the current behaviour is because `JPADatabase` uses Hibernate to generate an H2 in-memory database from your annotations, which is then reverse engineered by the jOOQ code generator - and this is what H2 makes of `GenerationType.IDENTITY`.

We should certainly recognise this pattern and not treat this sequence expression as a `defaultValue()` expression, but as a `identity()`

As a workaround, I can offer you post-process generated code, replacing `"defaultValue\(.*?".*?".*?\)"` by `"identity(true)"`

**Comment by lukaseder** [comment]
Another workaround is, of course, to generate the code from your PostgreSQL database directly...

**Comment by lukaseder** [comment]
Fixed with #6339
